app/models/audio_file_template.rb

```diff
@@ -20,4 +23,32 @@ def set_defaults
     self.length_minimum ||= 0
     self.length_maximum ||= 0
   end
+
+  def validate_audio_file(file)
+    [audio_length_errors(file), label_mismatch_errors(file)].join(' ').strip
+  end
+
+  private
+
+  def audio_length_errors(file)
+    length_errors = ''
+    shorter_than_min = file.length < length_minimum
+    longer_than_max = length_maximum != 0 && file.length > length_maximum
+
+    if shorter_than_min || longer_than_max
+      length_errors << "Audio file #{file.position} '#{file.label}' is " +
```

Comment:
Maybe a bit verbose, with the label in there multiple times.  And is the position necessary?  (Since the labels _should_ match between the template and the file).